### PR TITLE
remove old logger from local server, use slog and span http, add span_id and trace_sampled attrs to log

### PR DESCRIPTION
**Pull request by James-Pickett**
* uses new slogger throughout local server
* add helper func to traces to add trace info to request ctx
* adds trace_id to log shipping
* adds trace_sampled value to log shipping

relates to https://github.com/kolide/launcher/issues/1449

### CODE REVIEW

File: pkg/traces/traces.go [hunk 1]
```diff
@@ -20,6 +21,16 @@ const (
...
 )
 
+// StartHttpRequestSpan returns a copy of the request with a new context to include span info and span,
+// including information about the calling function as appropriate.
+// Standardizes the tracer name. The caller is always responsible for
+// ending the span. `keyVals` should be a list of pairs, where the first in the pair is a
+// string representing the attribute key and the second in the pair is the attribute value.
+func StartHttpRequestSpan(r *http.Request, keyVals ...interface{}) (*http.Request, trace.Span) {
+	ctx, span := StartSpan(r.Context(), keyVals...)
+	return r.WithContext(ctx), span
```
Comment (James-Pickett) [inline pkg/traces/traces.go:31]: I'm mostly sure this is the correct way to do this. I followed the otel code [here](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/net/http/otelhttp/v0.46.0/instrumentation/net/http/otelhttp/handler.go#L229), but I would not mind a second pair of eyes to make sure I'm doing it correctly. We could also use the otel included https stuff, but maybe it's over kill for our little servers?

File: ee/localserver/server.go [hunk 2]
```diff
@@ -62,28 +61,16 @@ const (
...
-	return func(s *localServer) {
-		s.logger = log.With(logger, "component", "localserver")
-	}
-}
-
-func New(k types.Knapsack, opts ...LocalServerOption) (*localServer, error) {
+func New(k types.Knapsack) (*localServer, error) {
 	ls := &localServer{
-		logger:                log.NewNopLogger(),
+		logger:                k.Slogger().With("component", "localserver"),
```
Comment (directionless) [inline ee/localserver/server.go:66]: I ight have called this `slogger`, but okay!

File: ee/localserver/server.go [hunk 3]
```diff
@@ -155,17 +142,29 @@ func (ls *localServer) LoadDefaultKeyIfNotSet() error {
 
 	serverRsaCertPem := k2RsaServerCert
 	serverEccCertPem := k2EccServerCert
+
+	ctx := context.TODO()
+	slogLevel := slog.LevelDebug
```
Comment (directionless) [inline ee/localserver/server.go:147]: Does anything change this?

**Review by RebeccaMahany — DISMISSED** [review]

File: ee/localserver/logging-handler.go [hunk 4]
```diff
@@ -22,7 +21,8 @@ func (ls *localServer) requestLoggingHandler(next http.Handler) http.Handler {
 		recorder := &statusRecorder{ResponseWriter: w, Status: 200}
 
 		defer func(begin time.Time) {
-			level.Debug(ls.logger).Log(
+			ls.logger.Log(r.Context(), slog.LevelDebug,
```
[the remark this answers is not in the document]
Comment (James-Pickett) [reply ee/localserver/logging-handler.go]: updated

**Review by RebeccaMahany — APPROVED** [review]

File: ee/localserver/server.go (hunk 2 above)
Comment (James-Pickett) [reply ee/localserver/server.go]: using slogger

File: ee/localserver/server.go (hunk 3 above)
Comment (directionless) [reply ee/localserver/server.go:147]: I think if you don't plan on implementing that, it's more confusing than helpful.

**Review by directionless — APPROVED** [review]
Woo

File: ee/localserver/server.go (hunk 3 above)
Comment (James-Pickett) [reply ee/localserver/server.go:147]: I will remove the var next time I touch this area.

**Pull request merged**